spec/unit/knife/core/windows_bootstrap_context_spec.rb

```diff
@@ -20,24 +20,20 @@
 require "chef/knife/core/windows_bootstrap_context"
 describe Chef::Knife::Core::WindowsBootstrapContext do
   let(:config) { {} }
-  let(:chef_config) { Chef::Config.save } # "dup" to a hash
+  let(:chef_config) { {} }
```

Comment:
Nothing broke when I changed this, so ✂️

Reply:
This makes me nervous since you're snipping out all the default values in Chef::Config (although perhaps I'm missing context here where chef_config != Chef::Config, but then the point still applies though where that design is probably fundamentally wrong).  If we're not testing against the actual Chef::Config here we have no confidence that it works with all the Chef::Config defaults set.

Reply:
It is Chef::Config where it gets used in lib/ https://github.com/chef/chef/blob/master/lib/chef/knife/bootstrap.rb#L518-L527

The other bootstrap context tests provide a few values by default, but then they get superseded in the majority of the tests: https://github.com/chef/chef/blob/master/spec/unit/knife/core/bootstrap_context_spec.rb#L25-L33

I was thinking along the same lines originally, and I wondered if that had something to do with the weird way the `chef_config` was being overwritten in most of the tests. I thought using an empty hash made more sense to prevent hidden dependencies that would cause problems if the user changed the defaults. Since it doesn't seem to make a difference in the tests as they exist now I'll change it back. 👍

Reply:
I changed the tests to just use `Chef::Config` and then add on to that if necessary for a more realistic testing scenario.